Wrappers.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#include "MusicMachine.h"
+#include "Computer.h"
+#include "Teacher.h"
+#include "ToneControl.h"
+#include "Classroom.h"
+/*#include "Teacher.h"
+#include "ToneControl.h"
+#include "MusicMachine.h"
+#include "Classroom.h"*/
+
+struct ComputerWrapper
+{
+    ComputerWrapper( Computer* ptr ) : ptrToComputer( ptr ) { }
+    ~ComputerWrapper() 
+    {
+        delete ptrToComputer;
+    }
+    Computer* ptrToComputer = nullptr;
+};
+
+struct TeacherWrapper
+{
+    TeacherWrapper( Teacher* ptr ) : ptrToTeacher(ptr) { }
+    ~TeacherWrapper()
+    {
+        delete ptrToTeacher;
+    }
+    Teacher* ptrToTeacher = nullptr;
+};
+struct ToneControlWrapper
+{
+    ToneControlWrapper( ToneControl* ptr ) : ptrToToneControl(ptr) { }
+    ~ToneControlWrapper()
+    {
+        delete ptrToToneControl;
+    }
+    ToneControl* ptrToToneControl = nullptr;
+};
+struct ClassroomWrapper
+{
+    ClassroomWrapper( Classroom* ptr ) : ptrToClassroom(ptr) { }
+    ~ClassroomWrapper()
+    {
+        delete ptrToClassroom;
+    }
+    Classroom* ptrToClassroom = nullptr;
+};
+struct MusicMachineWrapper
+{
+    MusicMachineWrapper( MusicMachine* ptr ) : ptrToMusicMachine(ptr) { }
+    ~MusicMachineWrapper()
+    {
+        delete ptrToMusicMachine;
+    }
+    MusicMachine* ptrToMusicMachine = nullptr;
+};
```

Comment:
```
 4) Put all of your Wrapper classes in a single Wrappers.h file
         if you implemented your wrapper class functions in-class, you'll need to move them to Wrappers.cpp
         
 
 5) NO IN-CLASS IMPLEMENTATION ALLOWED.
         the only exception is the existing Atomic.h and LeakedObjectDetector.h
```

Remember:
- if you are not **using** a pointer or a reference, you can FORWARD-DECLARE the type in the header files.
- You will need to `#include` the pointed-to/referenced type where it is used: in the cpp file.

- [ ] revise